shell/platform/android/io/flutter/plugin/editing/ImeSyncDeferringInsetsCallback.java

```diff
@@ -131,24 +126,24 @@ public WindowInsets onProgress(
       if (!matching) {
         return insets;
       }
+
+      // The IME insets include the height of the navigation bar. If the app isn't laid out behind
+      // the navigation bar, this causes the IME insets will be too large during the animation.
+      // To fix this, we subtract the systemBars bottom inset if the system UI flags for laying out
+      // behind the navigation bar aren't present.
+      int excludedInsets = 0;
+      int systemUiFlags = view.getWindowSystemUiVisibility();
+      if ((systemUiFlags & View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION) == 0
+          && (systemUiFlags & View.SYSTEM_UI_FLAG_HIDE_NAVIGATION) == 0) {
+        excludedInsets = insets.getInsets(WindowInsets.Type.systemBars()).bottom;
```

Comment:
I can't recall a specific reason, this might just be left over from when I was experimenting with different solutions. 

The status bar insets shouldn't cause any problems as I'm only using the bottom inset, but I agree that it probably makes sense to only get the navbar insets here. I'll give it a test when I get the chance and push up the change.